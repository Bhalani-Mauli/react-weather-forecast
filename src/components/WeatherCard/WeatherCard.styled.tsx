import styled from "styled-components";
import { FaMapMarkerAlt } from "react-icons/fa";
import { FlxRow, FlxCol } from "../CommonStyles/commonStyles.styles";

export const Container = styled.div`
  margin-top: 15px;
  max-width: 250px;
  border-radius: 5px;
  background-color: ${(props) => props.theme.colors.calmgreen};
  color: ${(props) => props.theme.colors.darkblue};
`;

export const CardBody = styled.div`
  padding: 15px;
`;

export const CardHeader = styled(FlxRow)`
  justify-content: center;
  font-size: ${(props) => props.theme.fontSizes.small};
`;

export const LocationIcon = styled(FaMapMarkerAlt)`
  margin: 4px;
  font-size: 17px;
`;

export const CardDate = styled.div`
  font-size: ${(props) => props.theme.fontSizes.xxSmall};
`;

export const RowResultDataWrapper = styled(FlxRow)`
  margin-top: 10px;
  justify-content: space-evenly;
  border-bottom: 1px solid;
  padding: 5px 0;
  background-color: #03c8a2;
`;

export const MainTemprature = styled(FlxCol)`
  align-items: center;
  margin-top: 20px;
`;

export const Unit = styled.sup`
  align-items: center;
  font-size: ${(props) => props.theme.fontSizes.xxSmall};
`;

export const Temprature = styled.span`
  font-size: ${(props) => props.theme.fontSizes.large};
`;

export const ImgLogo = styled.img`
  max-width: 40px;
`;
