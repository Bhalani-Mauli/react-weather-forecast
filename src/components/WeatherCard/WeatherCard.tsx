import moment from "moment";
import { FaTint, FaWind, FaRegClock } from "react-icons/fa";
import { TbGauge, TbSunrise, TbSunset } from "react-icons/tb";
import { City, FilteredWeatherType } from "types/api";
import { IconWithText } from "./components/IconWithText";
import { WeatherDataField } from "./components/WeatherDataField";
import {
  Container,
  CardBody,
  CardHeader,
  LocationIcon,
  CardDate,
  RowResultDataWrapper,
  MainTemprature,
  Temprature,
  Unit,
  ImgLogo,
} from "./WeatherCard.styled";

interface PropTypes {
  weatherData: FilteredWeatherType;
  cityData: City;
}
function WeatherCard({ weatherData, cityData }: PropTypes) {
  const { name, country, sunrise, sunset } = cityData;
  const { calculatedData, date, hourly } = weatherData;
  const { humdity, pressure, temp, windSpeed } = calculatedData;
  const isTodaysCard = moment().isSame(date, "day");
  return (
    <Container isTodaysCard={isTodaysCard}>
      <CardBody>
        <CardHeader>
          <LocationIcon />
          <div>
            {name},{country}
            <CardDate>{date}</CardDate>
          </div>
        </CardHeader>

        <MainTemprature>
          <Temprature>
            {temp}
            <Unit>°C</Unit>
          </Temprature>
        </MainTemprature>

        <RowResultDataWrapper>
          <IconWithText text={windSpeed} unit="km/h">
            <FaWind size={12} />
          </IconWithText>
          <IconWithText text={humdity} unit="%">
            <FaTint size={12} />
          </IconWithText>
          <IconWithText text={pressure} unit="hPa">
            <TbGauge size={12} />
          </IconWithText>
        </RowResultDataWrapper>

        <RowResultDataWrapper>
          <IconWithText text={sunrise}>
            <TbSunrise size={12} />
          </IconWithText>
          <IconWithText text={sunset}>
            <TbSunset size={12} />
          </IconWithText>
        </RowResultDataWrapper>

        {hourly?.map((i, index) => (
          <WeatherDataField
            key={`${index}-${i.time}`}
            name={i.time}
            unit="°C"
            value={i.temp.toFixed(0)}
          >
            <FaRegClock />
          </WeatherDataField>
        ))}
      </CardBody>
    </Container>
  );
}

export default WeatherCard;
