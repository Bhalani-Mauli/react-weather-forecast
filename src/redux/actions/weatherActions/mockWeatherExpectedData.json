{
  "cityData": {
    "coord": { "lat": 23.0333, "lon": 72.6167 },
    "country": "IN",
    "id": 1279233,
    "name": "Ahmedabad",
    "population": 3719710,
    "sunrise": "06:36",
    "sunset": "18:12",
    "timezone": 19800
  },
  "weatherData": [
    {
      "calculatedData": {
        "humdity": 68,
        "pressure": 1012,
        "temp": 23,
        "windSpeed": 3
      },
      "date": "2022-10-16",
      "hourly": [{ "temp": 23.02, "time": "21:00" }]
    },
    {
      "calculatedData": {
        "humdity": 40,
        "pressure": 1012,
        "temp": 28,
        "windSpeed": 3
      },
      "date": "2022-10-17",
      "hourly": [
        { "temp": 22.97, "time": "00:00" },
        { "temp": 24.32, "time": "03:00" },
        { "temp": 31.19, "time": "06:00" },
        { "temp": 33.37, "time": "09:00" },
        { "temp": 30.8, "time": "12:00" },
        { "temp": 27.68, "time": "15:00" },
        { "temp": 25.96, "time": "18:00" },
        { "temp": 24.59, "time": "21:00" }
      ]
    },
    {
      "calculatedData": {
        "humdity": 34,
        "pressure": 1010,
        "temp": 28,
        "windSpeed": 3
      },
      "date": "2022-10-18",
      "hourly": [
        { "temp": 23.28, "time": "00:00" },
        { "temp": 25.68, "time": "03:00" },
        { "temp": 31.95, "time": "06:00" },
        { "temp": 34.01, "time": "09:00" },
        { "temp": 31.52, "time": "12:00" },
        { "temp": 28.08, "time": "15:00" },
        { "temp": 26.13, "time": "18:00" },
        { "temp": 24.58, "time": "21:00" }
      ]
    },
    {
      "calculatedData": {
        "humdity": 34,
        "pressure": 1010,
        "temp": 28,
        "windSpeed": 3
      },
      "date": "2022-10-19",
      "hourly": [
        { "temp": 23.02, "time": "00:00" },
        { "temp": 25.33, "time": "03:00" },
        { "temp": 31.78, "time": "06:00" },
        { "temp": 34.23, "time": "09:00" },
        { "temp": 31.34, "time": "12:00" },
        { "temp": 27.51, "time": "15:00" },
        { "temp": 26.08, "time": "18:00" },
        { "temp": 24.45, "time": "21:00" }
      ]
    },
    {
      "calculatedData": {
        "humdity": 37,
        "pressure": 1010,
        "temp": 28,
        "windSpeed": 3
      },
      "date": "2022-10-20",
      "hourly": [
        { "temp": 23.34, "time": "00:00" },
        { "temp": 25.39, "time": "03:00" },
        { "temp": 31.31, "time": "06:00" },
        { "temp": 33.83, "time": "09:00" },
        { "temp": 31.75, "time": "12:00" },
        { "temp": 28.52, "time": "15:00" },
        { "temp": 26.4, "time": "18:00" },
        { "temp": 24.26, "time": "21:00" }
      ]
    },
    {
      "calculatedData": {
        "humdity": 43,
        "pressure": 1012,
        "temp": 29,
        "windSpeed": 5
      },
      "date": "2022-10-21",
      "hourly": [
        { "temp": 23.41, "time": "00:00" },
        { "temp": 26.26, "time": "03:00" },
        { "temp": 31.85, "time": "06:00" },
        { "temp": 34.05, "time": "09:00" },
        { "temp": 31.84, "time": "12:00" },
        { "temp": 28.66, "time": "15:00" },
        { "temp": 27.77, "time": "18:00" }
      ]
    }
  ]
}
