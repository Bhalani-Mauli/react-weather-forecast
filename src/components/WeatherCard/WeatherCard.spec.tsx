import { render } from "@utils/test-utils";
import WeatherCard from "./WeatherCard";
import mockWeatherExpectedData from "@redux/actions/weatherActions/mockWeatherExpectedData.json";

jest.mock("moment", () => {
  return () => jest.requireActual("moment")("2022-10-16T00:00:00.000Z");
});

describe("Weatherapp", () => {
  it("render the weather component ", () => {
    const props = {
      weatherData: mockWeatherExpectedData.weatherData[0],
      cityData: mockWeatherExpectedData.cityData,
      tempUnit: "C",
      windUnit: "km/h",
    };
    const { asFragment } = render(<WeatherCard {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
