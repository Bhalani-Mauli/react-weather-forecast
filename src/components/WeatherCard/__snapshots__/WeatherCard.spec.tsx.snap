// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weatherapp render the weather component  1`] = `
<DocumentFragment>
  <div
    class="sc-jrcTuL cIPuNW"
  >
    <div
      class="sc-kDvujY dXMnKB"
    >
      <div
        class="sc-bcXHqe sc-ipEyDJ iueWEx jZMNlT"
      >
        <svg
          class="sc-csuSiG fUXBmD"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 384 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
          />
        </svg>
        <div>
          Ahmedabad,India
          <div
            class="sc-eDWCr hAuRuh"
          >
            12.10.2022
          </div>
        </div>
      </div>
      <div
        class="sc-gswNZR sc-ksBlkl gQYmtx eajOYu"
      >
        <span
          class="sc-fnGiBr kGwgvO"
        >
          20
          <sup
            class="sc-hBxehG TwZeE"
          >
            °C
          </sup>
        </span>
        <div
          class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
        >
          <img
            class="sc-fEXmlR doywYA"
            src="http://openweathermap.org/img/wn/01n@2x.png"
          />
          <div
            class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
          />
          <div
            class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
          >
            Cloudy 
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe sc-bqWxrE iueWEx tGWlX"
      >
        <div
          class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
        >
          <svg
            fill="currentColor"
            height="12"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M156.7 256H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h142.2c15.9 0 30.8 10.9 33.4 26.6 3.3 20-12.1 37.4-31.6 37.4-14.1 0-26.1-9.2-30.4-21.9-2.1-6.3-8.6-10.1-15.2-10.1H81.6c-9.8 0-17.7 8.8-15.9 18.4 8.6 44.1 47.6 77.6 94.2 77.6 57.1 0 102.7-50.1 95.2-108.6C249 291 205.4 256 156.7 256zM16 224h336c59.7 0 106.8-54.8 93.8-116.7-7.6-36.2-36.9-65.5-73.1-73.1-55.4-11.6-105.1 24.9-114.9 75.5-1.9 9.6 6.1 18.3 15.8 18.3h32.8c6.7 0 13.1-3.8 15.2-10.1C325.9 105.2 337.9 96 352 96c19.4 0 34.9 17.4 31.6 37.4-2.6 15.7-17.4 26.6-33.4 26.6H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16zm384 32H243.7c19.3 16.6 33.2 38.8 39.8 64H400c26.5 0 48 21.5 48 48s-21.5 48-48 48c-17.9 0-33.3-9.9-41.6-24.4-2.9-5-8.7-7.6-14.5-7.6h-33.8c-10.9 0-19 10.8-15.3 21.1 17.8 50.6 70.5 84.8 129.4 72.3 41.2-8.7 75.1-41.6 84.7-82.7C526 321.5 470.5 256 400 256z"
            />
          </svg>
          <div
            class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
          />
          <div
            class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
          >
            22 
            <div
              class="sc-bcXHqe sc-eDvSVe iueWEx fDhrZd"
            >
              km/h
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
        >
          <svg
            fill="currentColor"
            height="12"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 352 512"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M205.22 22.09c-7.94-28.78-49.44-30.12-58.44 0C100.01 179.85 0 222.72 0 333.91 0 432.35 78.72 512 176 512s176-79.65 176-178.09c0-111.75-99.79-153.34-146.78-311.82zM176 448c-61.75 0-112-50.25-112-112 0-8.84 7.16-16 16-16s16 7.16 16 16c0 44.11 35.89 80 80 80 8.84 0 16 7.16 16 16s-7.16 16-16 16z"
            />
          </svg>
          <div
            class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
          />
          <div
            class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
          >
            22 
            <div
              class="sc-bcXHqe sc-eDvSVe iueWEx fDhrZd"
            >
              km/h
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
        >
          <svg
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <desc />
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <circle
              cx="12"
              cy="12"
              r="9"
            />
            <circle
              cx="12"
              cy="12"
              r="1"
            />
            <line
              x1="13.41"
              x2="16"
              y1="10.59"
              y2="8"
            />
            <path
              d="M7 12a5 5 0 0 1 5 -5"
            />
          </svg>
          <div
            class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
          />
          <div
            class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
          >
            1012 
            <div
              class="sc-bcXHqe sc-eDvSVe iueWEx fDhrZd"
            >
              hPa
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe sc-bqWxrE iueWEx tGWlX"
      >
        <div
          class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
        >
          <svg
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <desc />
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <path
              d="M3 17h1m16 0h1m-15.4 -6.4l.7 .7m12.1 -.7l-.7 .7m-9.7 5.7a4 4 0 0 1 8 0"
            />
            <line
              x1="3"
              x2="21"
              y1="21"
              y2="21"
            />
            <path
              d="M12 9v-6l3 3m-6 0l3 -3"
            />
          </svg>
          <div
            class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
          />
          <div
            class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
          >
            7:30 
          </div>
        </div>
        <div
          class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
        >
          <svg
            fill="none"
            height="12"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <desc />
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <path
              d="M3 17h1m16 0h1m-15.4 -6.4l.7 .7m12.1 -.7l-.7 .7m-9.7 5.7a4 4 0 0 1 8 0"
            />
            <line
              x1="3"
              x2="21"
              y1="21"
              y2="21"
            />
            <path
              d="M12 3v6l3 -3m-6 0l3 3"
            />
          </svg>
          <div
            class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
          />
          <div
            class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
          >
            6:30 
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe sc-gKPRtg iueWEx duKPgb"
      >
        <div
          class="sc-bcXHqe sc-iBYQkv iueWEx cSQWld"
        >
          <div
            class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
              />
            </svg>
            <div
              class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
            />
            <div
              class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
            >
              12:00 
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe sc-ftTHYK iueWEx LwgKS"
        >
          20 
          <div
            class="sc-bcXHqe sc-pyfCe iueWEx cbTTQz"
          >
            °C
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe sc-gKPRtg iueWEx duKPgb"
      >
        <div
          class="sc-bcXHqe sc-iBYQkv iueWEx cSQWld"
        >
          <div
            class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
              />
            </svg>
            <div
              class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
            />
            <div
              class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
            >
              3:00 
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe sc-ftTHYK iueWEx LwgKS"
        >
          22 
          <div
            class="sc-bcXHqe sc-pyfCe iueWEx cbTTQz"
          >
            °C
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe sc-gKPRtg iueWEx duKPgb"
      >
        <div
          class="sc-bcXHqe sc-iBYQkv iueWEx cSQWld"
        >
          <div
            class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
              />
            </svg>
            <div
              class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
            />
            <div
              class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
            >
              6:00 
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe sc-ftTHYK iueWEx LwgKS"
        >
          10 
          <div
            class="sc-bcXHqe sc-pyfCe iueWEx cbTTQz"
          >
            °C
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe sc-gKPRtg iueWEx duKPgb"
      >
        <div
          class="sc-bcXHqe sc-iBYQkv iueWEx cSQWld"
        >
          <div
            class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
              />
            </svg>
            <div
              class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
            />
            <div
              class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
            >
              9:00 
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe sc-ftTHYK iueWEx LwgKS"
        >
          20 
          <div
            class="sc-bcXHqe sc-pyfCe iueWEx cbTTQz"
          >
            °C
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe sc-gKPRtg iueWEx duKPgb"
      >
        <div
          class="sc-bcXHqe sc-iBYQkv iueWEx cSQWld"
        >
          <div
            class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
              />
            </svg>
            <div
              class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
            />
            <div
              class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
            >
              12:00 
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe sc-ftTHYK iueWEx LwgKS"
        >
          20 
          <div
            class="sc-bcXHqe sc-pyfCe iueWEx cbTTQz"
          >
            °C
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe sc-gKPRtg iueWEx duKPgb"
      >
        <div
          class="sc-bcXHqe sc-iBYQkv iueWEx cSQWld"
        >
          <div
            class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
              />
            </svg>
            <div
              class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
            />
            <div
              class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
            >
              3:00 
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe sc-ftTHYK iueWEx LwgKS"
        >
          22 
          <div
            class="sc-bcXHqe sc-pyfCe iueWEx cbTTQz"
          >
            °C
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe sc-gKPRtg iueWEx duKPgb"
      >
        <div
          class="sc-bcXHqe sc-iBYQkv iueWEx cSQWld"
        >
          <div
            class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
              />
            </svg>
            <div
              class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
            />
            <div
              class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
            >
              6:00 
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe sc-ftTHYK iueWEx LwgKS"
        >
          10 
          <div
            class="sc-bcXHqe sc-pyfCe iueWEx cbTTQz"
          >
            °C
          </div>
        </div>
      </div>
      <div
        class="sc-bcXHqe sc-gKPRtg iueWEx duKPgb"
      >
        <div
          class="sc-bcXHqe sc-iBYQkv iueWEx cSQWld"
        >
          <div
            class="sc-bcXHqe sc-dkrFOg iueWEx enraLd"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
              />
            </svg>
            <div
              class="sc-gswNZR sc-jSUZER gQYmtx ckBsnF"
            />
            <div
              class="sc-bcXHqe sc-hLBbgP iueWEx irXefn"
            >
              9:00 
            </div>
          </div>
        </div>
        <div
          class="sc-bcXHqe sc-ftTHYK iueWEx LwgKS"
        >
          20 
          <div
            class="sc-bcXHqe sc-pyfCe iueWEx cbTTQz"
          >
            °C
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
